### PR TITLE
Add work request function

### PR DESCRIPTION
**Pull request by wayneyu**
This function queries OKE work requests and logs them. The logs are captured by the function's invocation logs which connectorHub (https://github.com/Faire/platform-configs/pull/16676/files) will consume and forward to DD.

### CODE REVIEW

File: work-request-exporter/function/func.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import logging
+from datetime import datetime, timedelta, timezone
+import oci
+import os
+
+from fdk import response
+
+
+REGIONS = os.environ.get("REGIONS", "ca-toronto-1,us-ashburn-1")
+COMPARTMENT_IDS = os.environ.get("COMPARTMENT_IDS", "ocid1.compartment.oc1..aaaaaaaanydaqleheyo7hepeg6lsvplv7igvfarnomdigsk253h3mxkxjocq,ocid1.compartment.oc1..aaaaaaaaqzxstunqgzwnvfyrg5xoj4glcqqkygszmlljx4ke4epqiynce2pq")
```
Comment (bpoland) [inline work-request-exporter/function/func.py]: can we either dynamically get these or at least add a comment indicating what they correspond to?

File: work-request-exporter/function/func.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        "work_request_id": work_request_summary.id,
+        "error": error.message,
+        "time": work_request_summary.time_started.isoformat(),
+      } for error in errors]
+
+  return work_requests_errors
+
+if __name__ == "__main__":
+  get_work_requests_errors(
+    compartment_id="ocid1.compartment.oc1..aaaaaaaanydaqleheyo7hepeg6lsvplv7igvfarnomdigsk253h3mxkxjocq",
```
Comment (bpoland) [inline work-request-exporter/function/func.py]: same, how does this correspond to the IDs above?

File: work-request-exporter/function/func.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+name: work-request-exporter
+version: 0.0.47
+runtime: python
+build_image: fnproject/python:3.11-dev
+run_image: fnproject/python:3.11
+entrypoint: /python/bin/fdk /function/func.py handler
+memory: 256
+timeout: 120
+config:
+  COMPARTMENT_IDS: ocid1.compartment.oc1..aaaaaaaanydaqleheyo7hepeg6lsvplv7igvfarnomdigsk253h3mxkxjocq,ocid1.compartment.oc1..aaaaaaaaqzxstunqgzwnvfyrg5xoj4glcqqkygszmlljx4ke4epqiynce2pq
```
Comment (bpoland) [inline work-request-exporter/function/func.yaml]: same

File: work-request-exporter/function_setup.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+resource "null_resource" "Login2OCIR" {
+  count = local.user_image_provided ? 0 : 1
+  provisioner "local-exec" {
+    command = "echo '${var.oci_docker_password}' |  docker login ${local.oci_docker_repository} --username ${local.ocir_namespace}/${var.oci_docker_username} --password-stdin"
```
Comment (bpoland) [inline work-request-exporter/function_setup.tf:4]: lol this is kinda janky, is this how they recommend? Seems like it should be possible to have a docker credential helper

File: work-request-exporter/function_setup.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  lifecycle {
+    ignore_changes = [
+      defined_tags["Oracle-Tags.CreatedBy"],
+      defined_tags["Oracle-Tags.CreatedOn"],
+    ]
+  }
+}
+
+# ### build the function into a container image and push that image to the repository in the OCI Container Image Registry
+resource "null_resource" "FnImagePushToOCIR" {
```
Comment (bpoland) [inline work-request-exporter/function_setup.tf:28]: Is there really no way to use an existing image and build/push it ourselves outside of terraform? This all seems very odd to me

**Review by bpoland — COMMENTED** [review]
Sorry I don't have a ton of context on how this all works but left a few comments

File: work-request-exporter/function_setup.tf (hunk 4 above)
Comment (wayneyu) [reply work-request-exporter/function_setup.tf:4]: yea, this is from their offcial DD logs exporter setup

File: work-request-exporter/function_setup.tf (hunk 5 above)
Comment (wayneyu) [reply work-request-exporter/function_setup.tf:28]: But I guess the dockerfile and python function would need to be somewhere outside of platform-configs? I think we can definitely improve this if we have time

File: work-request-exporter/function/func.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+import logging
+from datetime import datetime, timedelta, timezone
+import oci
+import os
+
+from fdk import response
+
+
+REGIONS = os.environ.get("REGIONS", "ca-toronto-1,us-ashburn-1") # comma separated list of regions to export OKE cluster work requests errors
+COMPARTMENTS = os.environ.get("COMPARTMENTS", "ci,crafting") # comma separated list of compartment ids for which to export all OKE cluster work requests errors
```
Comment (bpoland) [inline work-request-exporter/function/func.py]:
```suggestion
COMPARTMENTS = os.environ.get("COMPARTMENTS", "ci,crafting") # comma separated list of compartment names for which to export all OKE cluster work requests errors
```

**Review by bpoland — APPROVED** [review]
don't love all the local-exec stuff but if it works then 🤷

**Pull request merged**